### DISCUSSION TITLE
External eeprom and sdcard

### DISCUSSION THREAD

**Discussion by xpertcrush** [OpenI6X/opentx#169] [Ideas]
Hi ajjjjj and team,

Is it possible to add external eeprom and sd card, so there's no problem with space and can build a full version of open tx. I can try adding/replacing eeprom if its possible?

**Comment by ajjjjjjjj** [comment]
Hi, moved to discussions.

One of the aims of this project is to make it work on basic hardware without any extensions so you can just flash and use it. 
Eeprom nor SDCard do not solve flash space issue and if by "full version" OpenTX you mean voice and audio then it needs more RAM as well. 
To give you some metrics how it is on 128kB of FLASH: some of the devices have 256kB and do not support OpenTX because official one starts at 512kB. The best we can do from my perspective is to create more dense code so more flash space is recovered.

If you want you may try to enable SDCard - I've done some research before and it needs about 24kB with standard FAT library but it can be replaced with Petite FAT, audio can be omited and so on.

Any contribution is very welcomed.
